python/cudf/cudf/core/frame.py

```diff
@@ -3500,42 +3443,15 @@ def _colwise_binop(
                         left_column = left_column.fillna(fill_value)
                     elif right_column.nullable:
                         right_column = right_column.fillna(fill_value)
+                else:
+                    if left_column.nullable:
+                        left_column = left_column.fillna(fill_value)
 
-            # For bitwise operations we must verify whether the input column
-            # types are valid, and if so, whether we need to coerce the output
-            # columns to booleans.
-            coerce_to_bool = False
-            if fn_apply in {"and", "or", "xor"}:
-                err_msg = (
-                    f"Operation 'bitwise {fn_apply}' not supported between "
-                    f"{left_column.dtype.type.__name__} and {{}}"
-                )
-                if right_column is None:
-                    raise TypeError(err_msg.format(type(None)))
-
-                try:
-                    left_is_bool = issubdtype(left_column.dtype, np.bool_)
-                    right_is_bool = issubdtype(right_column.dtype, np.bool_)
-                except TypeError:
-                    raise TypeError(err_msg.format(type(right_column)))
-
-                coerce_to_bool = left_is_bool or right_is_bool
-
-                if not (
-                    (left_is_bool or issubdtype(left_column.dtype, np.integer))
-                    and (
-                        right_is_bool
-                        or issubdtype(right_column.dtype, np.integer)
-                    )
-                ):
-                    raise TypeError(
-                        err_msg.format(right_column.dtype.type.__name__)
-                    )
+            # TODO: Disable logical and binary operators between columns that
+            # are not numerical using the new binops mixin.
```

Comment:
For your better tracking, this line might be better added to L211 of numerical.py?

Reply:
Thanks, I think this placement is fine for now though. I'm going to have to address this for all different column types (not just numerical), the mixin should make that easy since all operations become easy to define by opting in.